File: abstract-factory/c-sharp/MySQLDatabaseFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AbstractFactory
{
    // MySQL için ilişkili nesneler üretildi.
    class MySQLDatabaseFactory : IDatabaseFactory
    {
        public Command CreateCommand()
        {
            return new MySQLCommand();
        }

        public Connection CreateConnection()
        {
            return new MySQLConnection();
        }
    }
}

File: abstract-factory/go/mysql-database-factory.go
package main

// MySQL için ilişkili nesneler üretildi.
type mysqlDatabaseFactory struct{}

func NewMySQLDatabaseFactory() *mysqlDatabaseFactory {
	return &mysqlDatabaseFactory{}
}

func (f *mysqlDatabaseFactory) createConnection() Connection {
	return newMySQLConnection()
}

func (f *mysqlDatabaseFactory) createCommand() Command {
	return newMySQLCommand()
}
